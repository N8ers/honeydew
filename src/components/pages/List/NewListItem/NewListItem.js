import { useState } from "react"
import TextField from "@mui/material/TextField"
import Button from "@mui/material/Button"
import PropTypes from "prop-types"

function NewListItem(props) {
  const { addListItem } = props

  const [newListItem, setNewListItem] = useState("")

  const addNewListItem = (event) => {
    event.preventDefault()
    addListItem(newListItem)
    setNewListItem("")
  }

  return (
    <form onSubmit={addNewListItem} data-testid="newListItemForm">
      <TextField
        id="outlined-basic"
        variant="outlined"
        size="small"
        value={newListItem}
        data-testid="textField"
        onChange={(event) => setNewListItem(event.target.value)}
      />
      <Button
        variant="contained"
        type="submit"
        data-testid="addButton"
        style={{
          backgroundColor: "#1ea5a3",
          color: "white",
          fontWeight: "bold",
          marginLeft: "5px",
        }}
      >
        Add
      </Button>
    </form>
  )
}

NewListItem.propTypes = {
  addListItem: PropTypes.func,
}

export default NewListItem
