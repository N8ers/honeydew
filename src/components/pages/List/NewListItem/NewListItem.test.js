import { fireEvent, render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

import NewListItem from "./NewListItem"

describe("NewListItem.js", () => {
  test("Renders default contnet", async () => {
    const mock_addListItem = jest.fn()
    render(<NewListItem addListItem={mock_addListItem} />)
    const wrapper = screen.getByTestId("newListItemForm")
    expect(wrapper).toBeInTheDocument()
    expect(wrapper).toMatchSnapshot()
  })

  test("Add and submit a new item", () => {
    const mock_addListItem = jest.fn()
    render(<NewListItem addListItem={mock_addListItem} />)

    const inputField = screen.getByTestId("textField").querySelector("input")
    expect(inputField).toBeInTheDocument()

    fireEvent.change(inputField, { target: { value: "hi" } })
    expect(inputField.value).toBe("hi")

    const addButton = screen.getByTestId("addButton")
    fireEvent.click(addButton)

    expect(mock_addListItem).toHaveBeenCalledTimes(1)
    expect(mock_addListItem).toHaveBeenCalledWith("hi")

    expect(inputField.value).toBe("")
  })
})
