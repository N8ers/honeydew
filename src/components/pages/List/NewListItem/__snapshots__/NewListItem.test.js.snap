// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewListItem.js Renders default contnet 1`] = `
<form
  data-testid="newListItemForm"
>
  <div
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    data-testid="textField"
  >
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        id="outlined-basic"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-hdw1oc"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-testid="addButton"
    style="background-color: rgb(30, 165, 163); color: white; font-weight: bold; margin-left: 5px;"
    tabindex="0"
    type="submit"
  >
    Add
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</form>
`;
